jfr-mappers/src/main/java/com/newrelic/jfr/toevent/JITCompilationMapper.java

```diff
@@ -35,10 +35,13 @@
     var duration = event.getDuration();
     var attr = new Attributes();
     attr.put("desc", MethodSupport.describeMethod(event.getValue("method")));
-    attr.put("thread.name", event.getThread("eventThread").getJavaName());
     attr.put("duration", duration.toMillis());
     attr.put("succeeded", Workarounds.getSucceeded(event));
 
+    var threadId = event.getThread("eventThread");
+    if (threadId != null && threadId.getJavaName() != null) {
+      attr.put("thread.name", threadId.getJavaName());
+    }
```

Comment:
My other comments about NPE not being possible don't apply here since this code calls a nested accessor. However, if we want to retain null values rather than dropping the map entries when values are null, it might be updated to:
```suggestion
    var eventThread = event.getThread("eventThread");
    attr.put("thread.name", eventThread == null ? null : eventThread.getJavaName())
```

Reply:
Note that the initial issue reported was not an NPE, but rather "java.lang.IllegalArgumentException: Could not find field with name method". There is also a no-arg event.getThread() method - I'm not clear on the significance of the "eventThread" name being passed in. I'm unable to reproduce the original issue at the moment but will try again.

Reply:
I think what's going on is related to [comment here](https://github.com/newrelic/newrelic-jfr-core/pull/126#discussion_r548225280). 

Different versions of jvms may make different field names available, and the caller is supposed to take care to check if the field exists or risk `IllegalArgumentException` being thrown.